### PR TITLE
[CIApp] - Change CIEnvironmentValues from static to singleton.

### PR DESCRIPTION
**Pull request by tonyredondo**
@DataDog/apm-dotnet

### CODE REVIEW

**Comment by tonyredondo** [comment]
> LGTM. Were there any specific reasons that you preferred a singleton here?

@andrewlock found an issue that may related to this. Also I've just prefer the singleton pattern here, to not trigger the GirParser accidentaly

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2389 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f7ab92304bda9402c39acba7ceafd0b58059ec5f) will **not change** line coverage
:heavy_check_mark: Merging #2389 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f7ab92304bda9402c39acba7ceafd0b58059ec5f) will **not change** branch coverage
:no_entry: Merging #2389 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f7ab92304bda9402c39acba7ceafd0b58059ec5f) will will **increase** complexity by `2`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/f7ab92304bda9402c39acba7ceafd0b58059ec5f) | #2389       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `10395` / `14789` | `10351` / `14791` |          |
| Lines %   | `70%`      | `70%`      |  `0%` :heavy_check_mark:  |
| Branches  | `5341` / `7774` | `5313` / `7774` |          |
| Branches %| `69%`      | `68%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `8904`      | `8906`        |  `2`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=88938&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/88938/artifacts?artifactName=Code%20Coverage%20Report_88938&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2389](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=88967&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/88967/artifacts?artifactName=Code%20Coverage%20Report_88967&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/f7ab92304bda9402c39acba7ceafd0b58059ec5f) | #2389       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `70%`      | `70%`       |  `0%` :heavy_check_mark:  |
| Branches %| `69%`      | `68%`       |  `0%` :heavy_check_mark:  |
| Complexity| `8904`      | `8906`       |  `2` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.IIOServiceExecuteIntegrationBis](https://github.com/DataDog/dd-trace-dotnet/tree/0a56c3c0dba30f55e9359aca139109e131298415/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/IIOServiceExecuteIntegrationBis.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2.TestMethodRunnerExecuteIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/0a56c3c0dba30f55e9359aca139109e131298415/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs) | `-33%` :no_entry: | `-19%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Agent.CIAgentWriter](https://github.com/DataDog/dd-trace-dotnet/tree/0a56c3c0dba30f55e9359aca139109e131298415/tracer/src/Datadog.Trace/Ci/Agent/CIAgentWriter.cs) | `-14%` :no_entry: | `-21%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.NUnit.NUnitCompositeWorkItemSkipChildrenIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/0a56c3c0dba30f55e9359aca139109e131298415/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs) | `-8%` :no_entry: | `-8%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.ServerlessInstrumentation.AWS.LambdaNoParam](https://github.com/DataDog/dd-trace-dotnet/tree/0a56c3c0dba30f55e9359aca139109e131298415/tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaNoParam.cs) | `100%` :heavy_check_mark: | `100%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.ServerlessInstrumentation.AWS.LambdaOneParam](https://github.com/DataDog/dd-trace-dotnet/tree/0a56c3c0dba30f55e9359aca139109e131298415/tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaOneParam.cs) | `100%` :heavy_check_mark: | `100%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.ServerlessInstrumentation.AWS.LambdaTwoParams](https://github.com/DataDog/dd-trace-dotnet/tree/0a56c3c0dba30f55e9359aca139109e131298415/tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaTwoParams.cs) | `100%` :heavy_check_mark: | `100%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=88938&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/88938/artifacts?artifactName=Code%20Coverage%20Report_88938&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2389](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=88967&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/88967/artifacts?artifactName=Code%20Coverage%20Report_88967&api-version=6.0&%24format=zip)

File: tracer/src/Datadog.Trace/Ci/Agent/CIAgentWriter.cs [hunk 1]
```diff
@@ -65,10 +65,10 @@ public void WriteTrace(ArraySegment<Span> trace)
                 }
             }
 
-            foreach (var span in trace)
+            for (var i = trace.Offset; i < trace.Count + trace.Offset; i++)
```
Comment (lucaspimentel) [inline tracer/src/Datadog.Trace/Ci/Agent/CIAgentWriter.cs:68]: you found it! 😅

**Review by lucaspimentel — APPROVED** [review]

**Review by lucaspimentel — APPROVED** [review]

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #2389 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5): 
* 1 benchmarks are **faster**, with geometric mean 1.201
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`WriteAndFlushEnrichedTraces`|net472|852μs|3.35μs|13μs|0| 0|0|3.16&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|677μs|2.2μs|8.51μs|0| 0|0|2.57&nbsp;KB|
|#2389|`WriteAndFlushEnrichedTraces`|net472|892μs|3.81μs|14.8μs|0| 0|0|3.16&nbsp;KB|
|#2389|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|663μs|3.63μs|21.2μs|0| 0|0|2.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`SendRequest`|netcoreapp3.1|296μs|1.5μs|6.53μs|0.143| 0|0|19.67&nbsp;KB|
|#2389|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|#2389|`SendRequest`|netcoreapp3.1|302μs|1.34μs|5.03μs|0.151| 0|0|19.67&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`ExecuteNonQuery`|net472|1.59μs|4.34ns|16.2ns|0.0931| 0.000782|0|586&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`ExecuteNonQuery`|netcoreapp3.1|1.51μs|8.04ns|40.2ns|0.00868| 0|0|624&nbsp;B|
|#2389|`ExecuteNonQuery`|net472|1.63μs|9.1ns|57.5ns|0.0923| 0.000803|0|586&nbsp;B|
|#2389|`ExecuteNonQuery`|netcoreapp3.1|1.46μs|3.97ns|15.4ns|0.00884| 0|0|624&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`CallElasticsearch`|net472|2.34μs|13.5ns|99.9ns|0.124| 0|0|794&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`CallElasticsearch`|netcoreapp3.1|1.73μs|8.47ns|34.9ns|0.0106| 0|0|784&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`CallElasticsearchAsync`|net472|2.42μs|8.23ns|30.8ns|0.145| 0|0|931&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`CallElasticsearchAsync`|netcoreapp3.1|1.77μs|6.32ns|24.5ns|0.0122| 0|0|904&nbsp;B|
|#2389|`CallElasticsearch`|net472|2.32μs|11.8ns|51.4ns|0.124| 0|0|794&nbsp;B|
|#2389|`CallElasticsearch`|netcoreapp3.1|1.63μs|6.38ns|27.1ns|0.0107| 0|0|784&nbsp;B|
|#2389|`CallElasticsearchAsync`|net472|2.49μs|13ns|61.1ns|0.145| 0|0|931&nbsp;B|
|#2389|`CallElasticsearchAsync`|netcoreapp3.1|1.74μs|5.07ns|19.7ns|0.0124| 0|0|904&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Faster :tada: Same allocations :heavy_check_mark:



#### Faster :tada: in #2389

|  Benchmark         | base/diff | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.GraphQLBenchmark.ExecuteAsync&#8209;net472 | 1.201 | 3,166.23 | 2,635.99 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`ExecuteAsync`|net472|3.15μs|11.6ns|44.9ns|0.164| 0|0|1.05&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`ExecuteAsync`|netcoreapp3.1|2.11μs|10.2ns|42.2ns|0.0136| 0|0|1.02&nbsp;KB|
|#2389|`ExecuteAsync`|net472|2.64μs|7.9ns|29.6ns|0.165| 0|0|1.05&nbsp;KB|
|#2389|`ExecuteAsync`|netcoreapp3.1|1.97μs|8.93ns|33.4ns|0.0145| 0|0|1.02&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`SendAsync`|net472|6μs|26ns|101ns|0.352| 0|0|2.26&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`SendAsync`|netcoreapp3.1|4.44μs|15.7ns|60.7ns|0.0288| 0|0|2.16&nbsp;KB|
|#2389|`SendAsync`|net472|5.97μs|22.8ns|88.2ns|0.352| 0|0|2.26&nbsp;KB|
|#2389|`SendAsync`|netcoreapp3.1|4.63μs|23.3ns|107ns|0.0293| 0|0|2.16&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`EnrichedLog`|net472|3.39μs|14ns|52.4ns|0.225| 0|0|1.44&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`EnrichedLog`|netcoreapp3.1|3.07μs|10.5ns|40.6ns|0.0214| 0|0|1.52&nbsp;KB|
|#2389|`EnrichedLog`|net472|3.38μs|13.2ns|51ns|0.225| 0|0|1.44&nbsp;KB|
|#2389|`EnrichedLog`|netcoreapp3.1|3.21μs|18.1ns|122ns|0.0206| 0|0|1.52&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`EnrichedLog`|net472|293μs|1.09μs|4.07μs|0.437| 0.146|0|4.31&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`EnrichedLog`|netcoreapp3.1|232μs|807ns|3.13μs|0| 0|0|4.19&nbsp;KB|
|#2389|`EnrichedLog`|net472|292μs|1.38μs|7.69μs|0.43| 0.143|0|4.31&nbsp;KB|
|#2389|`EnrichedLog`|netcoreapp3.1|236μs|978ns|3.79μs|0| 0|0|4.19&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`EnrichedLog`|net472|7.97μs|42.8ns|287ns|0.501| 0|0|3.21&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`EnrichedLog`|netcoreapp3.1|6.63μs|36.3ns|208ns|0.0496| 0|0|3.58&nbsp;KB|
|#2389|`EnrichedLog`|net472|7.55μs|20.6ns|77.2ns|0.502| 0|0|3.21&nbsp;KB|
|#2389|`EnrichedLog`|netcoreapp3.1|6.55μs|36.4ns|224ns|0.0497| 0|0|3.58&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`SendReceive`|net472|2.36μs|9.8ns|38ns|0.157| 0|0|1&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`SendReceive`|netcoreapp3.1|1.94μs|9.25ns|35.8ns|0.0144| 0|0|1&nbsp;KB|
|#2389|`SendReceive`|net472|2.17μs|9.48ns|35.5ns|0.158| 0|0|1&nbsp;KB|
|#2389|`SendReceive`|netcoreapp3.1|1.96μs|8ns|31ns|0.0135| 0|0|1&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`EnrichedLog`|net472|6.43μs|31.2ns|132ns|0.287| 0|0|1.85&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`EnrichedLog`|netcoreapp3.1|5.43μs|29.6ns|167ns|0.0194| 0|0|1.47&nbsp;KB|
|#2389|`EnrichedLog`|net472|5.96μs|27.8ns|111ns|0.29| 0|0|1.85&nbsp;KB|
|#2389|`EnrichedLog`|netcoreapp3.1|5.1μs|25.5ns|111ns|0.02| 0|0|1.47&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`StartFinishSpan`|net472|909ns|4.17ns|16.7ns|0.0705| 0|0|449&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`StartFinishSpan`|netcoreapp3.1|878ns|3.7ns|14.3ns|0.0062| 0|0|448&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`StartFinishScope`|net472|1.08μs|3.05ns|11.4ns|0.0831| 0|0|530&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/34ed364f3147597f40ea923813a57430fcd02fa5)|`StartFinishScope`|netcoreapp3.1|1.03μs|4.87ns|18.8ns|0.00812| 0|0|568&nbsp;B|
|#2389|`StartFinishSpan`|net472|911ns|3.84ns|14.9ns|0.0707| 0|0|449&nbsp;B|
|#2389|`StartFinishSpan`|netcoreapp3.1|933ns|2.81ns|10.9ns|0.00643| 0|0|448&nbsp;B|
|#2389|`StartFinishScope`|net472|1.12μs|5.44ns|23.7ns|0.0833| 0|0|530&nbsp;B|
|#2389|`StartFinishScope`|netcoreapp3.1|1.04μs|4.88ns|19.5ns|0.00778| 0|0|568&nbsp;B|
</details>

**Pull request merged**